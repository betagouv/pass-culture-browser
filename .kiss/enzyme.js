import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
