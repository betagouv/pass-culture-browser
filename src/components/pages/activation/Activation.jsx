import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

import Error from './Error/Error'
import InvalidLink from './InvalidLink/InvalidLink'
import PasswordFormContainer from './PasswordForm/PasswordFormContainer'

const Activation = () => {
  return (
    <main className="logout-form-main">
      <Switch>
        <Route
          component={Error}
          exact
          path="/activation/error"
        />
        <Route
          component={InvalidLink}
          exact
          path="/activation/lien-invalide"
        />
        <Route
          component={PasswordFormContainer}
          path="/activation/:token"
        />
        <Redirect to="/activation/error" />
      </Switch>
    </main>
  )
}

export default Activation
