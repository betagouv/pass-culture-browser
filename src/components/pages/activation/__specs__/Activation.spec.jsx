import { mount } from 'enzyme'
import React from 'react'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router'

import getMockStore from '../../../../utils/mockStore'
import Activation from '../Activation'

describe('activation', () => {
  it('should render route for activating password when token adn email are given', () => {
    // when
    const mockStore = getMockStore({
      token: (
        state = {
          hasBeenChecked: false,
        }
      ) => state,
    })
    const wrapper = mount(
      <Provider store={mockStore}>
        <MemoryRouter initialEntries={['/activation/MEFA?email=beneficiary@example.com']}>
          <Activation />
        </MemoryRouter>
      </Provider>
    )

    // then
    const title = wrapper.find({ children: 'Pour commencer, choisis ton mot de passe.' })
    expect(title).toHaveLength(1)
  })

  it('should render error component when wrong route is given', () => {
    // given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/activation/error']}>
        <Activation />
      </MemoryRouter>
    )

    // then
    const title = wrapper.find({ children: 'Il semblerait que le lien cliqué soit incorrect.' })
    expect(title).toHaveLength(1)
  })

  it('should redirect to error page when current URLs does not match any mapped URLs', () => {
    // given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/activation/wrong-url']}>
        <Activation />
      </MemoryRouter>
    )

    // then
    const title = wrapper.find({ children: 'Il semblerait que le lien cliqué soit incorrect.' })
    expect(title).toHaveLength(1)
  })

  it('should render InvalidLink component when route is exactly /activation/lien-invalide', () => {
    // given
    const wrapper = mount(
      <MemoryRouter initialEntries={['/activation/lien-invalide']}>
        <Activation />
      </MemoryRouter>
    )

    // then
    const title = wrapper.find({ children: 'Le lien sur lequel tu as cliqué est invalide.' })
    expect(title).toHaveLength(1)
  })
})
