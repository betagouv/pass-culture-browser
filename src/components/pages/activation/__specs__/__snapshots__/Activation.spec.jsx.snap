// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | activation | Activation should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Activation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/activation/error"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/activation/lien-invalide"
        />
        <Route
          component={[Function]}
          path="/activation/:token"
        />
        <Redirect
          to="/activation/error"
        />
      </Switch>,
      "className": "logout-form-main",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/activation/error"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/activation/lien-invalide"
          />,
          <Route
            component={[Function]}
            path="/activation/:token"
          />,
          <Redirect
            to="/activation/error"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/activation/error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/activation/lien-invalide",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/activation/:token",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "to": "/activation/error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/activation/error"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/activation/lien-invalide"
          />
          <Route
            component={[Function]}
            path="/activation/:token"
          />
          <Redirect
            to="/activation/error"
          />
        </Switch>,
        "className": "logout-form-main",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/activation/error"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/activation/lien-invalide"
            />,
            <Route
              component={[Function]}
              path="/activation/:token"
            />,
            <Redirect
              to="/activation/error"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/activation/error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/activation/lien-invalide",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/activation/:token",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "to": "/activation/error",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
